pkg/rtc/participant.go

```diff
@@ -546,10 +547,21 @@ func (p *ParticipantImpl) SendSpeakerUpdate(speakers []*livekit.SpeakerInfo) err
 		return nil
 	}
 
+	var scopedSpeakers []*livekit.SpeakerInfo
+	for _, s := range speakers {
+		if p.IsSubscribedTo(s.Sid) {
```

Comment:
`subscribedTo` was keyed by identity, but `SpeakerInfo` only has SID. So, switching out the key to SID for the `subscribedTo` map which also necessitated the change in `SubscribedTrack` interface.